test/functional/pdo_sqlsrv/pdo_fetch_column_twice.phpt

```diff
@@ -0,0 +1,150 @@
+--TEST--
+Test fetchColumn twice in a row. Intentionally trigger various error messages.
+--DESCRIPTION--
+This is similar to sqlsrv_fetch_field_twice_data_types.phpt.
+--SKIPIF--
+<?php require('skipif_mid-refactor.inc'); ?>
+--FILE--
+<?php
+require_once("MsCommon_mid-refactor.inc");
+
+function fetchBeforeExecute($conn, $tableName, $inputs)
+{
+    try {
+        $tsql = "SELECT * FROM $tableName";
+        $stmt = $conn->prepare($tsql);
+        $row = $stmt->fetch(PDO::FETCH_ASSOC);
+        if ($row !== false) {
+            echo "fetchBeforeExecute: fetch should have failed before execute!\n";
+        }
+        $stmt->execute();
+        $row = $stmt->fetch(PDO::FETCH_NUM);
+
+        for ($i = 0; $i < count($inputs); $i++) {
+            if ($row[$i] !== $inputs[$i]) {
+                echo "fetchBeforeExecute: expected $inputs[$i] but got $row[$i]\n";
+            }
+        }
+
+        unset($stmt);
+    } catch (PDOException $e) {
+        var_dump($e->getMessage());
+    }
+}
+
+function fetchColumnTwice($conn, $tableName, $col, $input)
+{
+    try {
+        $tsql = "SELECT * FROM $tableName";
+        $stmt = $conn->query($tsql);
+        $result = $stmt->fetchColumn($col);
+        if ($result !== $input) {
+            echo "fetchColumnTwice (1): expected $input but got $result\n";
+        }
+        $result = $stmt->fetchColumn($col);
+        if ($result !== false) {
+            echo "fetchColumnTwice (2): expected the second fetchColumn to fail\n";
+        }
+
+        // Re-run the query with fetch style
+        $stmt = $conn->query($tsql, PDO::FETCH_COLUMN, $col);
+        $result = $stmt->fetch();
+        if ($result !== $input) {
+            echo "fetchColumnTwice (3): expected $input but got $result\n";
+        }
+        $result = $stmt->fetch();
+        if ($result !== false) {
+            echo "fetchColumnTwice (4): expected the second fetch to fail\n";
+        }
+        $result = $stmt->fetchColumn($col);
+        echo "fetchColumnTwice (5): expected fetchColumn to throw an exception\n";
+        unset($stmt);
+    } catch (PDOException $e) {
+        $error = '*There are no more rows in the active result set.  Since this result set is not scrollable, no more data may be retrieved.';
+
+        if (!fnmatch($error, $e->getMessage())) {
+            echo "Error message unexpected in fetchColumnTwice\n";
+            var_dump($e->getMessage());
+        }
+    }
+}
+
+function fetchColumnOutOfBound1($conn, $tableName, $col)
+{
+    try {
+        $tsql = "SELECT * FROM $tableName";
+        $stmt = $conn->query($tsql);
+        $result = $stmt->fetchColumn($col);
+        echo "fetchColumnOutOfBound1: expected fetchColumn to throw an exception\n";
+        unset($stmt);
+    } catch (PDOException $e) {
+        $error1 = '*General error: Invalid column index';
+        $error2 = '*An invalid column number was specified.';

```

Comment:
So either error might be returned?

Reply:
Yes, different error is returned depends on running with run-tests.php or not